### PR TITLE
Edge: announce notifications fired by main Edge process such as reading view availability

### LINKED ISSUE #8423: Microsoft Edge: Reading View notification is not announced
Hi,

Microsoft Edge comes with Reading view, a way to reformat the current page for smooth reading. However, NVDA does not announce certain notifications about it, including on pages where this view is not supported. Investigation shows this is tied to notification event introduced in IUIA5.

### Steps to reproduce:

1. Open Google with Edge.
2. Press Control+Shift+R to enable reading view.
> Please also remember to attach a zip of any files required to reproduce the issue.

### Expected behavior:
NVDA announces, "reading view is not available for this page".

### Actual behavior:
No announcement.

### System configuration:
NVDA next.15316 (installed), Windows 10 April 2018 Update.

### Other questions:
Reproducible: yes.
Did restart help: no.

### Technical details:
Certain notifications are sent via UiaRaiseNotificationEvent (see #7984 for details), and reading view announcement is one of them. At the moment NVDA checks to make sure focused control is part of the active application where the notification came from, but Edge uses at least two processes when run:

* MicrosoftEdge.exe: main process.
* MicrosoftEdgeCP.exe: content process.

Notifications are sent by the first process, thus failing the notification announcement criteria.

### Solution:
The Edge app module (microsoftedge.py) will gain notification event handler that'll announce all messages. This was tested on Windows 10 App Essentials add-on and was successful.

Thanks.

### PR DESCRIPTION
**Pull request by josephsl**
### Link to issue number:
Fixes #8423

### Summary of the issue:
Edge: notifications such as reading view availability is not announced.

### Description of how this pull request fixes the issue:
Certain notifications, including reading view availability are fired by main Edge process, not the content process. This means NvDA will ignore such notifications. Resolved by adding a dedicated app module for Edge main process that'll catch notifications and announce them, and propogate them no more.

### Testing performed:
Tested on Windows 10 April 2018 Update (UIA notification event is part of Fall Creators Update and later).

### Known issues with pull request:
None

### Change log entry:
Bug fixes: In Microsoft Edge, NVDA will announce notifications such as reading view availability and page load progress.

### CODE REVIEW

**Comment by josephsl** [comment]
Hi,

Ideally someone using Edge should also help out with reviewing this PR to catch bugs and such. Thanks.

File: source/appModules/microsoftedge.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import appModuleHandler
+import ui
+
+class AppModule(appModuleHandler.AppModule):
+
+	def event_UIA_notification(self, obj, nextHandler, displayString=None, **kwargs):
+		# #8423: even though content process is focused, notifications are fired by main Edge process.
+		# The base object will simply ignore this, so notifications must be announced here and no more.
+		ui.message(displayString)
```
Comment (leonardder) [inline source/appModules/microsoftedge.py]: Could you please end the file with an empty line?

File: source/appModules/microsoftedge.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+"""appModule for Microsoft Edge main process"""
+
+import appModuleHandler
+import ui
+
+class AppModule(appModuleHandler.AppModule):
+
+	def event_UIA_notification(self, obj, nextHandler, displayString=None, **kwargs):
+		# #8423: even though content process is focused, notifications are fired by main Edge process.
+		# The base object will simply ignore this, so notifications must be announced here and no more.
```
Comment (leonardder) [inline source/appModules/microsoftedge.py]: If I understand this correctly, these notifications will be announced regardless of what application has the foreground. It is imaginable that notifications are fired even though either this or a child process doesn't have focus. Could you somehow make sure that notifications are only announced within the context of Edge?

Furthermore, how to make sure that this appModule loads? Does it ever receive focus and if so, when?

**Comment by josephsl** [comment]
Hi, it should end with an empty line, but apparently it didn’t come through (I guess). I’ll add an extra empty line just in case. Thanks.

**Comment by josephsl** [comment]
Hi,

The Edge process loads when you run Edge and move to the app UI elements such as address omnibar, Settings window and what not.

As for limiting this to Edge context, that's next (thanks for reminding me about this).

Thanks,

**Comment by leonardder** [comment]
What will happen when a document window has focus, and you restart NVDA? 
Then, I assume the new appModule won't be loaded until you bring the 
process to the foreground in one of the ways you just mentioned.

**Review by LeonarddeR — DISMISSED** [review]

**Comment by josephsl** [comment]
Hi, I’d like to request a second look from @MichaelDCurran in hopes of making this one of the early test cases of a supposed new release process. Thanks.

**Review by michaelDCurran — APPROVED** [review]

**Pull request merged**